src/components/inputs/Select/index.js

```diff
@@ -1,123 +1,208 @@
-import React, { useState } from 'react';
+import React, { useState, useEffect } from 'react';
 import PropTypes from 'prop-types';
-import Select from 'react-select';
-import { withStyles } from '@material-ui/core';
-import { FormControl } from '../FormControl';
-import { InputLabel } from '../InputLabel';
+import TextField from '@material-ui/core/TextField';
+import Box from '@material-ui/core/Box';
+import Chip from '@material-ui/core/Chip';
+import Autocomplete from '@material-ui/lab/Autocomplete';
+import { makeStyles, CircularProgress } from '@material-ui/core';
 import { useSafeIntl } from '../../../utils/useSafeIntl';
 import { MESSAGES } from './messages';
-import { styles } from './styles';
 
-const SelectComponent = ({
+const useStyles = makeStyles(() => ({
+    chipLabel: {
+        marginTop: -2,
+    },
+    startAdornment: {
+        marginTop: -5,
+    },
+}));
+
+const SelectCustom = ({
     value,
     keyValue,
     label,
     errors,
     onChange,
     options,
+    touched,
     onBlur,
-    onFocus,
-    withMarginTop,
     multi,
     disabled,
     clearable,
-    isFocused,
-    searchable,
     required,
-    classes,
-    // noResultsText,
+    noOptionsText,
+    getOptionLabel,
+    getOptionSelected,
+    loading,
+    renderOption,
 }) => {
-    const [selectInputValue, setSelectInputValue] = useState('');
-    const hasErrors = errors.length > 0;
-    const classNames = hasErrors
-        ? [classes.select, classes.selectError]
-        : [classes.select];
+    const [selectedValue, setSelectedValue] = useState(null);
+    const [multiSelectedValue, setMultiSelectedValue] = useState([]);
     const intl = useSafeIntl();
-    const [focus, setFocus] = useState(isFocused);
+    const classes = useStyles();
+
+    const getOption = optionValue =>
+        options.find(o => `${o.value}` === `${optionValue}`);
+
+    useEffect(() => {
```

Comment:
I made the change requested by Son, but I'm actually not sure why this part is in a useEffect and not part of the handleChange? wouldn't that trigger a double render?

Reply:
This is a controlled input, the handleChange is only there to adapt the behaviour on change, useEffect is getting the correct value.